### PR TITLE
Fix building in Cygwin

### PR DESCRIPTION
**Pull request by gitplcc**
Fix building:
- Static and dynamic  link libraries (except Qt client).
- Drivers (except V4L2).
- Server.
- Unit and integration tests.

### CODE REVIEW

**Comment by gitplcc** [comment]
Sorry about forced-push. I forgot to rebase master before pull request.

File: libs/indicore/indicom.c [hunk 1]
```diff
@@ -331,7 +335,7 @@ double time_ns()
     ts.tv_sec = mts.tv_sec;
     ts.tv_nsec = mts.tv_nsec;
 #else
-    timespec_get(&ts, TIME_UTC);
+    #error "Unsupported platform"
```
Comment (pawel-soja) [inline libs/indicore/indicom.c:338]: Build fail for Windows CI:
```
D:\a\indi\indi\libs\indicore\indicom.c(338,1): fatal  error C1189: #error:  "Unsupported platform" [D:\a\indi\indi\build\libs\indicore\indicore.vcxproj]
  indidevapi.c
```

File: drivers/telescope/CMakeLists.txt [hunk 2]
```diff
@@ -194,7 +194,7 @@ install(TARGETS indi_skycommander_telescope RUNTIME DESTINATION bin)
 add_executable(indi_dsc_telescope
     dsc.cpp)
 
-target_link_libraries(indi_dsc_telescope indidriver AlignmentDriver)
+target_link_libraries(indi_dsc_telescope AlignmentDriver indidriver)
```
Comment (pawel-soja) [inline drivers/telescope/CMakeLists.txt:197]: What was the problem that you changed the linking order?

File: libs/indicore/indicom.c (hunk 1 above)
Comment (pawel-soja) [reply libs/indicore/indicom.c:338]:
```cmake
CHECK_FUNCTION_EXISTS(timespec_get HAVE_TIMESPEC_GET)
```
the result is a false negative. The function exists. It's possible that in `Visual Studio 17` the function is inline, either define or requires a system library.

Comment (pawel-soja) [reply libs/indicore/indicom.c:338]: Works:
```cmake
include(CheckSymbolExists)
check_symbol_exists(timespec_get time.h HAVE_TIMESPEC_GET)
```

File: drivers/telescope/CMakeLists.txt (hunk 2 above)
Comment (gitplcc) [reply drivers/telescope/CMakeLists.txt:197]: DLL linking was failing with original order. Inverting the order fixed it. I thought about chaging other similar 'target_link-libraries' sentences involving indidriver but I decided not to touch what was already working

File: libs/indicore/indicom.c (hunk 1 above)
Comment (gitplcc) [reply libs/indicore/indicom.c:338]: OK. I didn't read through till CheckFunctionExists module footnote. After reading it, I seems to me a good idea to replace every 'check_function_exists' to 'check_symbol_exists'.

**Comment by pawel-soja** [comment]
It looks okay.
I checked that there is no regression on the platforms in the search for functions.

### Ubuntu
branch: master
```
Looking for mremap - found
```
branch: this
```
Looking for mremap - found
Looking for timespec_get - found
Looking for clock_gettime - found
```
### MacOS
branch: master
```
Looking for mremap - not found
```
branch: this
```
Looking for mremap - not found
Looking for timespec_get - found
Looking for clock_gettime - found
```
### Windows
branch: master
```
Looking for mremap - not found
```
branch: this
```
Looking for mremap - not found
Looking for timespec_get - found
Looking for clock_gettime - not found
```

**Review by pawel-soja — APPROVED** [review]

**Comment by pawel-soja** [comment]
@knro It looks okay.

**Pull request merged**